site/js/MessagesActor.js

```diff
@@ -0,0 +1,425 @@
+          // strikethrough
+          toggleStyle('s')
+          c++ // skip charNext
+
+          continue
+        }
+
+        else if (char === '$' && charNext === '$' && currentToken !== 'latex') { startToken('latex'); c++; continue }
+        else if (char === '$' && charNext === '$' && currentToken === 'latex') { startToken('text'); c++; continue }
+
+        else if (char === '`' && currentToken !== 'code') { startToken('code'); continue }
+        else if (char === '`' && currentToken === 'code') { startToken('text'); continue }
+      }
+
+      buffer += char
+    }
+
+    styles = {}
+    startToken(null)
+    return el
+  }
+
+  // Builds the message content elements of a message. If the passed revision index
+  // is set to null, or is greater than the number of revisions, the most recent
+  // revision is used.
+  async buildMessageContent(message, revisionIndex = null) {
+    const { authorUsername } = message
+
+    if (!(revisionIndex in message.revisions)) {
+      revisionIndex = message.revisions.length - 1
+    }
+
+    if (revisionIndex < 0) {
+      revisionIndex = 0
+    }
+
+    const revision = message.revisions[revisionIndex]
+
+    const { text, signature } = revision
+
+    const el = document.createElement('div')
+    el.classList.add('message-revision-content')
+
+    const authorEl = document.createElement('div')
+    authorEl.classList.add('message-author')
+    authorEl.appendChild(document.createTextNode(authorUsername))
+
+    el.appendChild(authorEl)
+
+    const messageDate = new Date(Date.parse(message.date))
+    const pad = value => value.toString().padStart(2, '0')
+
+    const time = document.createElement('time')
+    time.setAttribute('datetime', messageDate.toISOString())
+    time.appendChild(document.createTextNode(
+      `${pad(messageDate.getHours())}:${pad(messageDate.getMinutes())}`
+    ))
+    el.appendChild(time)
+
+    const contentEl = this.formatMessageText(text)
+    contentEl.classList.add('message-content')
+    el.appendChild(contentEl)
+
+    /*
+    if (processPGP) {
+      // TODO: Re-write this code to work with user IDs rather than usernames.
+      if (signature) {
+        if (authorUsername in publicKeyDictionary === false) {
+          el.appendChild(document.createTextNode(' (Signed, but this user is not in your public key dictionary)'))
+        } else {
+          const verified = await openpgp.verify({
+            message: openpgp.cleartext.readArmored(signature),
+            publicKeys: openpgp.key.readArmored(publicKeyDictionary[authorUsername]).keys
+          })
+
+          if (verified.signatures[0].valid) {
+            el.appendChild(document.createTextNode(' (Verified)'))
+          } else {
+            el.appendChild(document.createTextNode(' (FORGED? Signature data provided, but sign did not match)'))
+          }
+        }
+      } else {
+        el.appendChild(document.createTextNode(' (No signature data)'))
+      }
+    }
+    */
```

Comment:
Noted. As discussed, this is fine :tm: for now, but I'd love to get this working again myself eventually!